### PR TITLE
feat: discover server IPs on registration, emit server events

### LINKED ISSUE #71: machine Spec doesn't have any IP information
Sidero:

```
status:
    bootstrapReady: true
    conditions:
    - lastTransitionTime: "2020-09-18T16:13:09Z"
      status: "True"
      type: Ready
    - lastTransitionTime: "2020-09-18T16:08:15Z"
      status: "True"
      type: BootstrapReady
    - lastTransitionTime: "2020-09-18T16:13:09Z"
      status: "True"
      type: InfrastructureReady
    infrastructureReady: true
    lastUpdated: "2020-09-18T16:13:09Z"
    nodeRef:
      apiVersion: v1
      kind: Node
      name: pxe-2
      uid: e6c0e820-2287-4e8d-95a1-76c1e3a541de
    observedGeneration: 3
    phase: Running
```

while for AWS/GCP we can have `.spec.addresses`:

```
MASTER_IP=$(${KUBECTL} --kubeconfig /tmp/e2e/docker/kubeconfig get machine -o go-template --template='{{range .status.addresses}}{{if eq .type "ExternalIP"}}{{.address}}{{end}}{{end}}' ${FIRST_CP_NODE})
```

### PR DESCRIPTION
**Pull request by smira**
At the moment, IPs are only discovered during the registration process
which is not sufficient, but more complete solution will be to add some
support to Talos for that.

Some events are emitted for servers as PoC. Registration, power
management, wipe, allocation are covered.

Fixes #71

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: app/metal-controller-manager/internal/server/server.go [hunk 1]
```diff
@@ -99,31 +111,117 @@ func (s *server) CreateServer(ctx context.Context, in *api.CreateServerRequest)
...
+
+	patchHelper, err := patch.NewHelper(obj, s.c)
 	if err != nil {
 		return nil, err
 	}
 
-	c, err := client.NewClient(config)
+	obj.Status.IsClean = true
+
+	if err := patchHelper.Patch(ctx, obj); err != nil {
```
Comment (smira) [inline app/metal-controller-manager/internal/server/server.go:127]: this might help with test flakiness

**Comment by smira** [comment]
Sample events:

```
Events:
  Type     Reason               Age   From                       Message
  ----     ------               ----  ----                       -------
  Normal   Server Registration  17m   sidero-server              Server auto-registered via API.
  Normal   Server Wipe          17m   sidero-server              Server got wiped via agent.
  Warning  Server Management    17m   sidero-controller-manager  Server powered off.
  Normal   Server Allocation    16m   caps-controller-manager    Server marked as allocated for metalMachine "management-cluster-workers-hnpzb"
  Warning  Server Management    16m   caps-controller-manager    Server powered on.
  Normal   Server Allocation    10m   caps-controller-manager    Server marked as unallocated
  Warning  Server Management    10m   sidero-controller-manager  Server power cycled.
  Normal   Server Allocation    10m   caps-controller-manager    Server marked as allocated for metalMachine "management-cluster-cp-dhq7f"
```

**Review by rsmitty — APPROVED** [review]

**Pull request merged**